File: poo/Cpp/Espelho/Espelho.cpp
#include <iostream>
#include <string>
#include <algorithm>

int main() {
    std::string a, b_str;
    int b_int;
    std::cin >> a >> b_str;
    std::reverse(a.begin(), a.end());
    if (a == b_str) {
        std::cout << "espelho" << std::endl;
    } else {
        std::cout << "nao espelho" << std::endl;
    }
    return 0;
}

File: poo/Java/Espelho/Espelho.java
import java.util.Scanner;

public class Espelho {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String a = scanner.next();
        String b = scanner.next();
        String aReversed = new StringBuilder(a).reverse().toString();
        if (aReversed.equals(b)) {
            System.out.println("espelho");
        } else {
            System.out.println("nao espelho");
        }
        scanner.close();
    }
}